node/src/protocol/cryptography.rs

```diff
@@ -73,7 +73,16 @@ impl CryptographicProtocol for GeneratingState {
         tracing::info!("progressing key generation");
         let mut protocol = self.protocol.write().await;
         loop {
-            let action = protocol.poke()?;
+            let action = match protocol.poke() {
+                Ok(action) => action,
+                Err(err) => {
+                    drop(protocol);
+                    if let Err(err) = self.protocol.refresh().await {
```

Comment:
yup, but the same parameters are supplied to the protocol initialization as when it was first created. It failing is very unlikely since we would've stopped way earlier if the parameters were incorrect